OneCommunityGlobal/HighestGoodNetworkApp
# TimeLogger for BMdashboard

# Description
Implementing "Time Logger" Function for bmdashboard.
http://localhost:3000/bmdashboard/timelog

## Related PRS (if any):
This PR should be tested along with [Backend 1041
](https://github.com/OneCommunityGlobal/HGNRest/pull/1041)

## Main changes explained:
This PR is for implementing the followings:

URL: “/bmdashboard/timelog”

By selecting a specific project, the manager can view the users assigned to the project.
The top bar shows the name of the user. The color may change according to the user’s role. For example, light blue for volunteers, orange for core team, light green for managers and mentors, and red for owner.
The “start”, “stop” or “clear” buttons are the same as the buttons used for time log in the phase one system.
A select dropdown for the manager to monitor. The value here will be the default/pre-select value passed to the time log form to save. 

![image](https://github.com/user-attachments/assets/ff4b48ad-ad61-4e9c-b62e-7797405d55b5)



## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to `http://localhost:3000/bmdashboard/timelog`, you may need to login again.
6. select a project to test
7. test start/pause/stop/clear time log for each member

## Screenshots or videos of changes:

![image](https://github.com/user-attachments/assets/8f430184-0554-4260-9d94-f32fe1af3ef9)

https://github.com/user-attachments/assets/d654b9e5-0a59-4177-8492-74678cbcdec3



## Function implemented by now:
1. retrieve project and member information from server to display member card
2. Stop/Pause/Stop/Clear time log
3. Set card color according to member role

## Functions to be implemented further
1. retrieve task name to display on the member's card
    --> may need to think about separate signaling to retrieve task information because of database fetch latency
2. update time log information to the database
3. some cosmetic improvement
